### ISSUE TITLE
Bug: [@typescript-eslint/keyword-spacing] import type is broken

### ISSUE THREAD

**Issue by Sonahit** [typescript-eslint/typescript-eslint#6093]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=JYWwDg9gTgLgBDAnmApnA3nAKigzvAXzgDMoIQ4AiACxhjEoG4AoZgYwgDtcIAbFAHS8IAcwAUOfAEpGQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Aa0UIHcB7aACa1kxAIZlmAc3RRE0aP0jgwAXxAqgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
import type { Test } from "http";
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/keyword-spacing": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Everything is OK

### Actual Result

TypeError: Cannot read properties of undefined (reading 'after')
Occurred while linting <input>:1
Rule: "@typescript-eslint/keyword-spacing"
    at ImportDeclaration[importKind=type] (https://typescript-eslint.io/sandbox/index.js:20:791664)
    at https://typescript-eslint.io/sandbox/index.js:20:258768
    at https://typescript-eslint.io/sandbox/index.js:20:249285
    at Array.forEach (<anonymous>)
    at Object.emit (https://typescript-eslint.io/sandbox/index.js:20:249273)
    at NodeEventGenerator$1.applySelector (https://typescript-eslint.io/sandbox/index.js:4:122933)
    at NodeEventGenerator$1.applySelectors (https://typescript-eslint.io/sandbox/index.js:4:123235)
    at NodeEventGenerator$1.enterNode (https://typescript-eslint.io/sandbox/index.js:4:123326)
    at CodePathAnalyzer$1.enterNode (https://typescript-eslint.io/sandbox/index.js:4:56574)
    at https://typescript-eslint.io/sandbox/index.js:20:259026

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `X.Y.Z` |
| `@typescript-eslint/parser`            | `5.44` |
| `TypeScript`                           | `4.8.4` |
| `ESLint`                               | `8.15.0` |

**Comment by bradzacher** [comment]
@Sonahit please use the search before filing issues
